output/cloud/expv2/output.go

```diff
@@ -164,9 +165,6 @@ func (o *Output) periodicInvoke(d time.Duration, callback func()) {
 
 func (o *Output) collectSamples() {
 	samples := o.GetBufferedSamples()
-	if len(samples) < 1 {
-		return
-	}
```

Comment:
Even if those assumptions are valid, I don't see how an early return if there's no work to be done would hurt. If nothing else it would simplify understanding this logic without being aware of these nuances.

But it's up to you :slightly_smiling_face:

Reply:
> I don't see how an early return if there's no work to be done would hurt.

Then the code is not obvious and I think we need to arrange it in another way. The reason why I removed it is because we need to flush the SampleBuffer + the collector's buffer. If we return we can't execute the drain of the collect's buffer. Do you think it makes sense?

The alternative is doing something one of the following in `StopWithTestError`:

```
output.collectSamples()
output.collector.collectSamples(nil)
```
this has the downside to call two times the `collector.CollectSamples` method, so another way could be:

```
output.collector.collectSamples(o.GetBufferedSamples())
```

Reply:
OK, I was misunderstanding where the flush is happening.

I still don't have a full grasp of the code here, but that's on me, so I don't think refactoring would help. :slightly_smiling_face: